### PR TITLE
feat: prevent the creation of deployments if there is already an active deployment with the same constructor parameters

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --fast

**Review by kjaskiewiczz — COMMENTED** [review]
~~I think we need to update API docs too~~
EDIT: sorry, I only briefly looked at the PR and thought we'll return 409 (which is undocumented), but we're returning 400;
I don't feel like 400 Bad Request is the best error code in this case.
IMO 409 will do better (some people are using 303, which feels a bit exotic?)
github is returning 422 in similar cases, and for me 422 was always a return code indicating that I'm giving up on http rfc ;) (still I'm using it from time to time)
@alfrunes @merlin-northern any opinions?

**Comment by tranchitella** [comment]
> ~I think we need to update API docs too~ EDIT: sorry, I only briefly looked at the PR and thought we'll return 409 (which is undocumented), but we're returning 400; I don't feel like 400 Bad Request is the best error code in this case. IMO 409 will do better (some people are using 303, which feels a bit exotic?) github is returning 422 in similar cases, and for me 422 was always a return code indicating that I'm giving up on http rfc ;) (still I'm using it from time to time) @alfrunes @merlin-northern any opinions?

look at the latest commit

File: docs/management_api.yml [hunk 1]
```diff
@@ -38,6 +38,10 @@ responses:
     description: Invalid Request.
     schema:
       $ref: "#/definitions/Error"
+  ConflictError: # 409
+    description: Conflict
```
Comment (kjaskiewiczz) [inline docs/management_api.yml:42]: What about being more specific here?
Like `An active deployment with the exact same parameters already exists`?

**Review by kjaskiewiczz — APPROVED** [review]
Looks good to me, but maybe it's a good idea to wait for @alfrunes or @merlin-northern opinion on the status code we're returning here?

File: docs/management_api.yml (hunk 1 above)
Comment (tranchitella) [reply docs/management_api.yml]: Updated

**Review by alfrunes — APPROVED** [review]

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --fast

**Pull request merged**